frontend/src/components/pages/teacher/DistributeAssessmentPage.tsx

```diff
@@ -123,14 +128,22 @@ const DistributeAssessmentPage = (): React.ReactElement => {
             Next
           </Button>
         ) : (
-          <Button
-            leftIcon={<PaperPlaneOutlineIcon />}
-            minWidth="10"
-            onClick={() => console.log("Distribute assessment.")}
-            variant="primary"
-          >
-            Distribute
-          </Button>
+          <>
+            {startDate && endDate && teacherId && (
+              <DistributeAssessmentButton
+                testSessionRequest={{
+                  test: testId,
+                  teacher: teacherId,
+                  school: "639151a4d46e8c002a49f8d6",
+                  class: classId,
+                  accessCode: generateAccessCode(),
```

Comment:
This will regenerate the access code every time this component renders, which react can choose to do at any point. Is that fine?

Reply:
u right, dats probs bad